Code/src/main/java/nl/utwente/group10/ui/components/Connection.java

```diff
@@ -1,98 +1,182 @@
 package nl.utwente.group10.ui.components;
 
+import java.util.Optional;
+
 import javafx.beans.value.ChangeListener;
 import javafx.beans.value.ObservableValue;
-import javafx.fxml.FXMLLoader;
-import javafx.fxml.Initializable;
-
-import java.io.IOException;
-import java.net.URL;
-import java.util.ResourceBundle;
+import javafx.geometry.Point2D;
 
 /**
- * This class represents a connection between two different FunctionBlocks. The
- * output of one FunctionBlock will be used as input for another FunctionBlock
+ * This is a ConnectionLine that also stores a startAnchor and an endAnchor to
+ * keep track of origin points of the Line.
+ * 
+ * For Lines that connect inputs and outputs of Blocks see Connection.
  */
-public class Connection extends Line implements ChangeListener<Number> , Initializable {
-
-	/** The Block that inputs data into this connection */
-	private Block input;
-	/** The Block that we output data into from this connection */
-	private Block output;
-	/** The argument field of the FunctionBlock that we are outputting data into */
-	private int outputarg;
-	/** The fxmlLoader responsible for loading the fxml.*/
-	private FXMLLoader fxmlLoader;
-
-	/**
-	 * Method that creates a new instance of this class along with it's visual
-	 * representation. 
-	 * @param from Anchor to start from.
-	 * @param toAnchor Anchor to end at.
-	 * @return a new instance of this class
-	 * @throws IOException
-	 */
-	public Connection(OutputAnchor from, InputAnchor to) throws IOException {
-		fxmlLoader = new FXMLLoader(getClass().getResource("/ui/Connection.fxml"));
-		fxmlLoader.setRoot(this);
-		fxmlLoader.setController(this);
-
-		input = from.getBlock();
-		output = to.getBlock();
-
-		input.layoutXProperty().addListener(this);
-		input.layoutYProperty().addListener(this);
-
-		output.layoutXProperty().addListener(this);
-		output.layoutYProperty().addListener(this);
+public class Connection extends ConnectionLine implements
+		ChangeListener<Number> {
+	/** Starting point of this Line that can be Anchored onto other objects */
+	private OutputAnchor startAnchor;
+	/** Ending point of this Line that can be Anchored onto other objects */
+	private InputAnchor endAnchor;
 
+	public Connection(){
+	}
```

Comment:
Deze thread is het met mij eens:
http://stackoverflow.com/questions/5286348/how-can-a-java-class-have-no-no-arg-constructor